src/main/java/org/eclipse/microprofile/starter/validation/PackageValidator.java

```diff
@@ -41,10 +41,9 @@
     public void validate(FacesContext facesContext, UIComponent uiComponent, Object value) throws ValidatorException {
 
         PackageNameValidator validator = retrieveInstance(PackageNameValidator.class);
-        String name = value.toString().replaceAll("-", ".");
-        if (!validator.isValidPackageName(name)) {
+        if (!validator.isValidPackageName(value.toString())) {
             FacesMessage msg =
-                    new FacesMessage("Field validation failed.",
+                    new FacesMessage(uiComponent.getId() + " field validation failed.",
```

Comment:
id should be internal and not necessary reflecting the label of the field (but in our case, id is the same as the label used for the field)

Better is to use label property.

java: `((HtmlInputText)uiComponent).getLabel()`
xhtml: add `label="groupId"`  on the <p:inputText>